project/cmake/scripts/ios/ArchSetup.cmake

```diff
@@ -9,15 +9,12 @@ set(SYSTEM_DEFINES -D_REENTRANT -D_FILE_DEFINED -D_FILE_OFFSET_BITS=64 -D_LARGEF
                    -D__STDC_CONSTANT_MACROS)
 set(PLATFORM_DIR linux)
 set(CMAKE_SYSTEM_NAME Darwin)
-if(WITH_ARCH)
-  set(ARCH ${WITH_ARCH})
+
+if(ARCH STREQUAL arm OR ARCH STREQUAL aarch64)
```

Comment:
Shouldn't it still be armv7 and arm64 for iOS?

There's one thing I think you didn't take into account in this PR: `ARCH` is included in the ending of binary addons (see https://github.com/xbmc/xbmc/blob/master/xbmc/addons/kodi-addon-dev-kit/include/kodi/libXBMC_addon.h#L64 and https://github.com/xbmc/xbmc/blob/master/project/cmake/scripts/common/Macros.cmake#L151 and https://github.com/xbmc/xbmc/blob/master/configure.ac#L581). If you change `ARCH` in CMake, everything will compile fine and even packaging might work, but then addons will just fail to load.

The cleanest solution would be to specify the binary addon ending in cmake (and autootools for the time being) as an additional variable and pass it to both `libXBMC_addon.h` and to the macro that builds the addon libs. It might be also a good idea to revise if the lib endings could be changed to the new ARCH var - but since these endings are used in a few more scripts, and some platforms may need special names, I'm not sure how feasible this is.
@wsnipex: do you have a better idea? btw: should this be v18?